### DISCUSSION TITLE
Is it possible to serialize an empty object using a macro?

### DISCUSSION THREAD

**Discussion by harry961209** [nlohmann/json#3036] [Q&A]
`class Test: public AbstractTestModule
{
public:
    
    struct InParam
    {
        //-- User input parameters here !!!
        // ...

        //-- Add all parameters to json
        _JSON_(InParam, ); **// Issue**
    } InParam;

    struct OutParam
    {
        //-- Default output parameters
        string result = ""; // **Need to serialize** 

        //-- User output parameters here !!!
        // ...

        //-- Add all parameters to json
        _JSON_(OutParam, result);
    } OutParam;

    //-- Add all parameters to json
    _JSON_(Test, InParam, OutParam);
}`
I want to serialize "Test" class without no member variable InParam, OutParam class.
plz answer me!!

**Comment by nlohmann** [comment]
I don't know what `_JSON` is doing. But https://json.nlohmann.me/features/arbitrary_types/ could be helpful for you.
